packages/react/src/hooks/useExperiment.ts

```diff
@@ -29,7 +29,7 @@ export function useExperiment(key: string, defaultValue = false): boolean {
   const { gates } = useContext(GateContext) ?? {};
   const xp = gates?.store?.experiments ?? {};
 
-  const hashedKey = createHash(key);
+  const hashedKey = createHash(key, String(gates?.salt));
```

Comment:
We need to do better here, gates should always be defined. We need to improve our types

Reply:
Definitely, mainly because core has a fallback value defined.